driver/src/transport.rs

```diff
@@ -1,79 +1,123 @@
+use isahc::prelude::{Body, Request, Response};
+use isahc::{Error as HttpError, HttpClient, ResponseExt};
+use log::{debug, warn};
+use serde::Deserialize;
 use serde_json::Value;
+use std::fmt::{self, Debug, Formatter};
+use std::str::FromStr;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
 
-/// A `Transport` wrapper that logs RPC messages
-#[derive(Clone, Debug)]
-pub struct LoggingTransport<T> {
-    inner: T,
-    level: Level,
+/// An HTTP transport implementation with timeout and logging.
+#[derive(Clone)]
+pub struct HttpTransport(Arc<HttpTransportInner>);
+
+struct HttpTransportInner {
+    url: String,
+    client: HttpClient,
+    id: AtomicUsize,
 }
 
-impl<T> LoggingTransport<T>
-where
-    T: Transport,
-{
-    /// Create a new `LoggingTranport` from an underlying transport and log
-    /// level.
-    pub fn new(inner: T, level: Level) -> Self {
-        LoggingTransport { inner, level }
+impl HttpTransport {
+    /// Creates a new HTTP transport with settings.
+    pub fn new(url: impl Into<String>, timeout: Duration) -> Result<HttpTransport, DriverError> {
+        let client = HttpClient::builder()
+            .timeout(timeout)
+            // NOTE: This is needed as curl with try to upgrade to HTTP/2 which
+            //   causes a HTTP 400 error with Ganache.
+            .version_negotiation(VersionNegotiation::http11())
+            .build()?;
+
+        Ok(HttpTransport(Arc::new(HttpTransportInner {
+            url: url.into(),
+            client,
+            id: AtomicUsize::default(),
+        })))
     }
 }
 
-impl<T> Transport for LoggingTransport<T>
-where
-    T: Transport,
-{
-    type Out = LoggingFuture<T::Out>;
+impl HttpTransportInner {
+    /// Performs an HTTP POST with the given data.
+    async fn post(&self, data: String) -> Result<(Response<Body>, String), HttpError> {
+        let http_request = Request::post(&self.url)
+            // NOTE: This is needed as Parity clients will respond with a HTTP
+            //   error when no content type is provided.
+            .header("Content-Type", "application/json")
+            .body(data)?;
+        let mut response = self.client.send_async(http_request).await?;
+        let content = response.text()?;
 
-    fn prepare(&self, method: &str, params: Vec<Value>) -> (RequestId, Call) {
-        self.inner.prepare(method, params)
+        Ok((response, content))
     }
 
-    fn send(&self, id: RequestId, request: Call) -> Self::Out {
-        log!(
-            self.level,
-            "sending request ID {}: {}",
-            id,
-            serde_json::to_string(&request).expect("request is invalid JSON")
-        );
-        LoggingFuture {
-            inner: self.inner.send(id, request),
-            id,
-            level: self.level,
+    /// Execute an HTTP JSON RPC request.
+    async fn execute(self: Arc<Self>, id: RequestId, request: Call) -> Result<Value, Web3Error> {
```

Comment:
nit: as this transport might be reused by other parts of the codebase where we make requests, maybe it's a good idea to name this with rpc in mind (e.g. `execute_rpc`)?